### ISSUE TITLE
Fordulókör Implementálása

### ISSUE THREAD

**Issue by AbyssPrince** [SzFMV2020-Tavasz/AutomatedCar-A#55]
VFB-ról (Virtual Function Bus) a kormány állását le kell kérdezni. A kerekek ez alapján forduljanak, ezt egy érték reprezentálja fokokban (double) -180tól +180ig (azért reális értékék inkább -90től +90ig, de ezek a határok is már elég extrémek ugyebár).

- Ez az érték kerüljön be egy helyi mezőbe vagy objektumba.
- Ez az érték kerüljön beírásra a VFB-ba

**Comment by varszegimarcell** [comment]
Rendkívül fontos az aktuális váltóállás figyelembevétele is.

**Comment by pintergreg** [comment]
Ez a 180 fokos kerékelfordulás rendkívül extrém. Határozottan emlékszem hogy gondolkodtak olyanon a városi parkolási problémák feloldásán ügyeskedők, hogy mi lenne ha a kerekek 90 fokban elfordulnának és akkor bárminemű manőverezés nélkül beállhatna az autó a parkolóba. Legjobb tudomásom szerint nincs ilyen jármű a piacon...

[Itt](https://gamedev.stackexchange.com/questions/50022/typical-maximum-steering-angle-of-a-real-car) 65 fokot írnak, én [példában](https://szfmv2020-tavasz.github.io/handout/physics.html#kanyarod%C3%A1s) 60-nal számoltam. 

@varszegimarcell Itt a hátramenetre gondolsz?

**Comment by varszegimarcell** [comment]
> @varszegimarcell Itt a hátramenetre gondolsz?

Így van.

> Ez a 180 fokos kerékelfordulás rendkívül extrém. Határozottan emlékszem hogy gondolkodtak olyanon a városi parkolási problémák feloldásán ügyeskedők, hogy mi lenne ha a kerekek 90 fokban elfordulnának és akkor bárminemű manőverezés nélkül beállhatna az autó a parkolóba. Legjobb tudomásom szerint nincs ilyen jármű a piacon...

Szerintem ez a 180 fokos kerékszög valahogy a kormányállás szögével keveredett itt össze, a kerék nyilván nem fog ennyit forogni.

**Comment by AbyssPrince** [comment]
> Szerintem ez a 180 fokos kerékszög valahogy a kormányállás szögével keveredett itt össze, a kerék nyilván nem fog ennyit forogni.

Nem, egyszerűen csak a matematikailag lehetséges értékkészletet írtam fel amiben reprezentáljuk a kerék forgásának mértékét. Nyilván irreális érték a plusz/mínusz 180, de mint értékkészlet ez az elméleti lehetséges minimum és maximum (+181 már -179-el egyenlő). Pont amiatt írtam oda, hogy gyakorlatilag nem lesz ennyi, mert a fizikailag nem terjed ki ekkora szögre a kerékfordulás.

**Comment by AbyssPrince** [comment]
Egyébként meg gondolkozom azon, hogy a kerékállás szögét a már amúgy is implementált vektorirányban (egységvektor) adom meg, amiből lekérdezhető radiánban, fokban, gradiánban... stb...

**Comment by varszegimarcell** [comment]
El kezdtem dolgozni a feladaton, de jelenleg nem kapok kormányszöget, így nincs mivel számolnom. 

Team1_dev jó lenne ha bekerülne a masterre. 
@SzFMV2020-Tavasz/team-a1
